src/include/test/unit_api_feature_vector.cc

```diff
@@ -110,7 +110,7 @@ TEST_CASE("api: FeatureVector dimension", "[api]") {
   CHECK(dimension(FeatureVector(Vector<int>{1, 2, 3})) == 3);
 }
 
-using TestTypes = std::tuple<float, uint8_t, int32_t, uint32_t, uint64_t>;
+using TestTypes = std::tuple<float, int32_t, uint32_t, uint64_t>;
```

Comment:
What does this change fix?

Reply:
Good question. `std::uniform_int_distribution` can not use 1 byte size variables in MSVC. The `randomize` function uses this distribution. I don't know how to fix this other way than to delete it.

Reply:
So no `char, uint8_t, int8_t` and friends are allowed.

Reply:
![1708530796_grim](https://github.com/TileDB-Inc/TileDB-Vector-Search/assets/5787996/418bb63c-af84-44d3-9d00-2d5dcc4cf39c)

Reply:
Hum.  Well it has been working.  Is this just an MSVC issue?  If so, address the issue by initializing the vector in some other way -- e.g., fill it random numbers generated in some other way, or with the digits of pi, or something else when in MSVC and the type is `uint8_t`..  `uint8_t` is an important type for vector search so we need to be able to test reading and writing it.

In general, making tests pass (even compilation) by removing the offending instance from the test suite is very bad practice.

In fact, I would claim this case did exactly what it is supposed to do -- it found a problem -- even if happened at compilation time rather than run time.  Again, when a unit test fails, it is telling you to fix the code, not the test.

Unit tests exist to tell you when something has gone amiss when you change your code (or your compilation environment).

(And good catch, Paris.)